src/SignalR/clients/csharp/Http.Connections.Client/src/HttpConnection.cs

```diff
@@ -75,14 +74,16 @@ public override IDuplexPipe Transport
         /// If the connection was created with <see cref="HttpConnectionOptions.SkipNegotiation"/> set to <c>true</c>
         /// then the connection ID will be <c>null</c>.
         /// </remarks>
-        public override string ConnectionId
+        public override string? ConnectionId
         {
+#pragma warning disable CS8764 // Nullability of return type doesn't match overridden member (possibly because of nullability attributes).
```

Comment:
Should ConnectionId be made nullable on ConnectionContext?

Reply:
I really think that people should be able to rely on `ConnectionContext.ConnectionId` being non-null. I'm thinking this should just throw if someone tries to access this property before `StartAsync()` completes.

This implementation of `ConnectionContext.ConnectionId` is already a bit odd in that its setter throws, so I don't think throwing from the getter when the connection hasn't started is a big deal.

Reply:
> people should be able to rely on `ConnectionContext.ConnectionId` being non-null

It's not just if you access it before starting, if you skip negotiation then this will be null as well.